pkg/velerorunner/backup.go

```diff
@@ -178,3 +190,60 @@ func (t *Task) getReplicatedBackup() (*velero.Backup, error) {
 
 	return nil, nil
 }
+
+// Delete the running restic pod in velero namespace
+// This function is used to get around mount propogation requirements
+func (t *Task) bounceResticPod() error {
+	client, err := t.getSourceClient()
+	if err != nil {
+		return err
+	}
+
+	// Set two vars to determine if we have bounced a pod and it is has become
+	// ready or if we need to delete the pod
+	var podDeleted = false
+	var podBounced = false
+	// Loop until pod has been deleted and is `Running`
+	for podBounced == false {
+		t.Log.Info(
+			"Waiting for restic pod to restart.")
+		// List all pods in velero namespace
+		list := corev1.PodList{}
+		err = client.List(
+			context.TODO(),
+			&k8sclient.ListOptions{
+				Namespace: VeleroNamespace,
+			},
+			&list)
+		if err != nil {
+			return err
+		}
+
+		// Loop through all pods in velero namespace
+		for _, pod := range list.Items {
+			if strings.HasPrefix(pod.Name, resticPodPrefix) {
```

Comment:
Recommend using `continue` to avoid the deep nesting.
```
if !strings.HasPrefix(pod.Name, resticPodPrefix) {
    continue
}
```
and
```
if pod.Status.Phase != corev1.PodRunning {
    continue
}

Reply:
+1

Reply:
Done, thanks!